antarest/study/output/output_service.py

```diff
@@ -393,80 +404,106 @@ def export_task(notifier: ITaskNotifier) -> TaskResult:
+                    TableExportFormat.PARQUET,
+                    data.columns,
+                    data.filter,
+                    file_path,
+                    transform_columns_headers=False,
+                    mc_years=data.years,
                 )
-            except Exception as e:
-                self._file_transfer_manager.fail(export_id, str(e))
-                raise
-
-        task_id = self._task_service.add_task(
-            export_task,
-            export_name,
-            task_type=TaskType.EXPORT,
-            ref_id=study_id,
-            progress=None,
-            custom_event_messages=None,
-        )
-
-        return FileDownloadTaskDTO(file=export_file_download.to_dto(), task=task_id)
+                # Wait for the aggregation to end
+                self._task_service.await_task(task_id)
+
+                # Aggregation can fail (for instance, when asking renewables values and no cluster exists)
+                # If so, we shouldn't raise to keep backward compatibility
+                task = self._task_service.status_task(task_id)
+                if task.status != TaskStatus.COMPLETED:
+                    file_path.unlink(missing_ok=True)
+                    continue
+
+                file_paths.append(file_path)
+
+            # Once they all ended, build the final response
+            intermediary_dict: dict[str, Any] = {}
+            # We're opening the parquet files chunk by chunk to avoid flooding memory
+            for dataframe in yield_dataframes_from_parquet(file_paths, []):
+                # Convert the dataframe in the right response
+                column_type_name = LINK_COL if data.type == StudyDownloadType.LINK else AREA_COL
+                df_columns = dataframe.columns.tolist()
+                df_columns.remove(column_type_name)
+                df_columns.remove(MCYEAR_COL)
+                for object_name, object_group in dataframe.groupby(column_type_name):
+                    assert isinstance(object_name, str)
+                    element_name = object_name
+                    if data.type == StudyDownloadType.LINK:
+                        element_name = "^".join(element_name.split(" - "))
+
+                    for year, year_group in object_group.groupby(MCYEAR_COL):
+                        variables_list = []
+                        for col in df_columns:
+                            # We're using arrays to avoid flooding memory
+                            output_data = year_group[col].to_numpy()
+                            splitted_col = col.split(" % ")
+                            name, unit = splitted_col[0], splitted_col[1]
+                            variables_list.append({"name": name, "unit": unit or " ", "data": output_data})
+
+                        intermediary_dict.setdefault(element_name, {}).setdefault(str(year), []).extend(variables_list)
+
+            response = MatrixAggregationResultDTO.model_validate(
```

Comment:
Not sure why we go through dictionaries adn `model_validate` instead of just using the pydantic models from the start ?